### DISCUSSION TITLE
Ability to choose entry type when choosing to create new entry

### DISCUSSION THREAD

**Discussion by angrybrad** [craftcms/cms#9429] [Ideas]
> Created by: Mark Bowen (markbowen@helitorque.com) on 2015/05/14 10:44:03 +0000
> Votes at time of UserVoice import: 3

-----------------------------------

Would be nice to be able to go to the Entries section of the CP and when you hover over New it would have a drop-down arrow where you could select the entry-type (if any exist) and then you would be taken to creating a new entry with that entry-type already selected instead of going to a new entry and then having to select the entry-type.

Can see this being very useful for clients to ensure they don't always just get the first entry-type in the list and forget to choose the correct one.

**Comment by KatieMFritz** [comment]
Yes! I assume this would involve updating the URL to include the entry type, which would make it possible to add dashboard buttons to create a new entry with a particular non-default type.

**Comment by brandonkelly** [comment]
@KatieMFritz Already possible with a `typeId` query param ;)

**Comment by KatieMFritz** [comment]
![bitmoji](https://render.bitstrips.com/v2/cpanel/ee47f314-e678-4ac2-b3dc-5fa399e731c8-3921b123-ba4b-40f5-b21b-39ab43205fc3-v1.png?transparent=1&palette=1&width=246)

Thanks @brandonkelly!

**Comment by khalwat** [comment]
Ugh no wonder I didn't see it in the method params... it's done in this way:

```php
        // Get the entry type
        // ---------------------------------------------------------------------

        // Override the entry type?
        $typeId = $request->getParam('typeId');

        if (!$typeId) {
            // Default to the section's first entry type
            $typeId = $variables['entry']->typeId ?? $variables['section']->getEntryTypes()[0]->id;
        }
```
